Date.cpp update breaks compilation on Solaris

Per https://www.r-project.org/nosvn/R.check/r-patched-solaris-sparc/Rcpp-00check.html:

```
/opt/csw/bin/g++ -I/home/ripley/R/gcc/include -DNDEBUG -I../inst/include/ -I/opt/csw/include -I/usr/local/include    -fPIC  -O2 -mcpu=niagara2  -c Date.cpp -o Date.o
Date.cpp: In function ‘std::tm* Rcpp::timesub(const time_t*, int_fast32_t, const Rcpp::state*, std::tm*)’:
Date.cpp:1352:14: error: ‘struct std::tm’ has no member named ‘tm_gmtoff’
         tmp->tm_gmtoff = offset;
              ^
*** Error code 1
make: Fatal error: Command failed for target `Date.o'
```

Ditto for the other Solaris build.

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,11 @@
+2017-01-17  Dirk Eddelbuettel  <edd@debian.org>
+
+        * DESCRIPTION (Version, Date): Roll minor version and Date
+
+        * inst/include/Rcpp/config.h (RCPP_DEV_VERSION): Roll minor
+
+        * src/Date.cpp (Rcpp): Do not access tm_gmtoff on Solaris
+
 2017-01-14  Dirk Eddelbuettel  <edd@debian.org>
 
         * DESCRIPTION: Release 0.12.9

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,7 +1,7 @@
 Package: Rcpp
 Title: Seamless R and C++ Integration
-Version: 0.12.9
-Date: 2017-01-14
+Version: 0.12.9.1
+Date: 2017-01-17
 Author: Dirk Eddelbuettel, Romain Francois, JJ Allaire, Kevin Ushey, Qiang Kou,
  Nathan Russell, Douglas Bates and John Chambers
 Maintainer: Dirk Eddelbuettel <edd@debian.org>

--- a/inst/include/Rcpp/config.h
+++ b/inst/include/Rcpp/config.h
@@ -30,7 +30,7 @@
 #define RCPP_VERSION Rcpp_Version(0,12,9)
 
 // the current source snapshot
-#define RCPP_DEV_VERSION RcppDevVersion(0,12,9,0)
+#define RCPP_DEV_VERSION RcppDevVersion(0,12,9,1)
 
 #endif
 

--- a/src/Date.cpp
+++ b/src/Date.cpp
@@ -1347,7 +1347,7 @@ struct tzhead {
             idays -= ip[tmp->tm_mon];
         tmp->tm_mday = (int) (idays + 1);
         tmp->tm_isdst = 0;
-#if ! (defined(__MINGW32__) || defined(__MINGW64__))
+#if ! (defined(__MINGW32__) || defined(__MINGW64__) || defined(__SUNPRO_CC))
 //#ifdef HAVE_TM_GMTOFF
         tmp->tm_gmtoff = offset;
 #endif

